presto-main/src/main/java/com/facebook/presto/operator/InMemoryJoinHash.java

```diff
@@ -168,7 +167,7 @@ public final long getNextJoinPosition(long currentJoinPosition, int probePositio
     private long getNextJoinPositionFrom(int startJoinPosition, int probePosition, Page allProbeChannelsPage)
     {
         long currentJoinPosition = startJoinPosition;
-        while (filterFunctionPresent && currentJoinPosition != -1 && !applyFilterFilterFunction(Ints.checkedCast(currentJoinPosition), probePosition, allProbeChannelsPage.getBlocks())) {
+        while (currentJoinPosition != -1 && !applyFilterFilterFunction(Ints.checkedCast(currentJoinPosition), probePosition, allProbeChannelsPage)) {
```

Comment:
Instead of doing these multiple checkedCasts, I would do it once like this:
```java
        int currentJoinPosition = Ints.checkedCast(startJoinPosition);
        while (currentJoinPosition != -1 && !applyFilterFilterFunction(currentJoinPosition, probePosition, allProbeChannelsPage)) {
            currentJoinPosition = positionLinks[currentJoinPosition];
        }
        return currentJoinPosition;